### PR TITLE
Eliminate JS errors from functional tests

### PR DESCRIPTION
**Pull request by seanf**
Based on revised code from http://stackoverflow.com/a/4411986/14379

Also removing `setTimeout()` support: the `timeoutCounter` is not used, and
  the wrapper code is causing problems.

This should eliminate the server error during `EditHomePageTest.editPageCode` and `EditProjectAboutTest.addAboutPageDetails`: `Uncaught TypeError: func is not a function`

### CODE REVIEW

File: zanata-war/src/main/java/org/zanata/action/AjaxCounterBean.java [hunk 1]
```diff
@@ -34,40 +34,27 @@
...
-            "            }\n" +
-            "        };\n" +
-            "        this._send.apply(this, arguments);\n" +
+            "    if (xhr.active === undefined) {\n" +
+            "        xhr.active = 0;\n" +
+            "        var pt = xhr.prototype;\n" +
+            "        var _send = pt.send;\n" +
+            "        pt.send = function() {\n" +
+            "            xhr.active++;\n" +
+            "            this.addEventListener('readystatechange', function(e) {\n" +
```
Comment (carlosmunoz) [inline zanata-war/src/main/java/org/zanata/action/AjaxCounterBean.java]: Adding the event listener here could mean we add a listener for every fired request, so each request will decrement the counter as many times as requests have been fired in total.

**Comment by carlosmunoz** [comment]
:+1:

**Comment by seanf** [comment]
@carlosmunoz Turns out we have a lot of JS warnings I hadn't noticed. (Some of them probably because RichFaces is using jQuery 2 by accident.) I've changed the test failure trigger to errors, not warnings.

**Comment by seanf** [comment]
@carlosmunoz Mind looking this over again?

**Comment by carlosmunoz** [comment]
:+1:

**Comment by djansen-redhat** [comment]
@zanata-jenkins retest this please

**Comment by seanf** [comment]
@djansen-redhat Looks like this passed.

**Comment by djansen-redhat** [comment]
:white_check_mark: Thank you Jenkins

**Comment by seanf** [comment]
Correction: this pull request does not in fact abort tests in case of JS errors, because `logLogs()` is called with `false`.

**Pull request merged**